src/galois/_fields/_gf2.py

```diff
@@ -99,6 +141,243 @@ def __init_subclass__(cls) -> None:
+            else:
+                # matrix-matrix multiplication
+                output = GF2.Zeros(final_shape)
+                for i in range(b_packed_shape[-1]):
+                    dot_product = (a & b.view(np.ndarray)[:, i]).view(np.ndarray)
+                    output[:, i] = self._dot_product_reduce(dot_product)
+                output_axis = a._axis
+
+            output = field._view(np.packbits(output.view(np.ndarray), axis=output_axis))
+            output._axis = output_axis
+            output._axis_count = final_shape[output_axis]
+
+        return output
+
+
+class concatenate_bitpacked(Function):
+    """Concatenates matrices together"""
+
+    def __call__(self, arrays: list[Array], **kwargs):
+        # TODO: Should we only unpack arrays that have column counts %8 != 0 and concatenate those first?
+        unpacked_arrays = []
+        for array in arrays:
+            verify_isinstance(array, self.field)
+            unpacked_arrays.append(np.unpackbits(array))
+
+        unpacked = np.concatenate(unpacked_arrays, **kwargs)
+        return np.packbits(unpacked)
+
+
+class inv_bitpacked(Function):
+    """
+    Computes the inverse of the square matrix.
+    """
+
+    def __call__(self, A: Array) -> Array:
+        verify_isinstance(A, self.field)
+        # if not (A.ndim == 2 and A.shape[0] == A.shape[1]):
+        #     raise np.linalg.LinAlgError(f"Argument 'A' must be square, not {A.shape}.")
+
+        n = A.shape[0]
+        I = self.field.Identity(n, dtype=A.dtype)
+
+        # Concatenate A and I to get the matrix AI = [A | I]
+        AI = np.concatenate((A, I), axis=-1)
+
+        # Perform Gaussian elimination to get the reduced row echelon form AI_rre = [I | A^-1]
+        AI_rre, _ = row_reduce_jit(self.field)(AI, ncols=n)
+
+        # The rank is the number of non-zero rows of the row reduced echelon form
+        rank = np.sum(~np.all(AI_rre[:, 0:n] == 0, axis=1))
+        if not rank == n:
+            raise np.linalg.LinAlgError(
+                f"Argument 'A' is singular and not invertible because it does not have full rank of {n}, "
+                f"but rank of {rank}."
+            )
+
+        A_inv = AI_rre[:, -n:]
+
+        return A_inv
+
+
+def not_implemented(*args, **kwargs):
+    # TODO: Add a better error message about limited support
+    return NotImplemented
+
+
+class UFuncMixin_2_1_BitPacked(UFuncMixin):
+    """
+    A mixin class that provides explicit calculation arithmetic for GF(2).
+    """
+
+    def __init_subclass__(cls) -> None:
+        super().__init_subclass__()
+        cls._add = add_ufunc_bitpacked(cls, override=np.bitwise_xor)
+        cls._negative = negative_ufunc(cls, override=np.positive)
+        cls._subtract = subtract_ufunc_bitpacked(cls, override=np.bitwise_xor)
+        cls._multiply = multiply_ufunc_bitpacked(cls, override=np.bitwise_and)
+        cls._reciprocal = not_implemented  # reciprocal(cls)
```

Comment:
What happens if `np.reciprocal()` is called on a `GF2BP` instance?

Reply:
These have been implemented now.